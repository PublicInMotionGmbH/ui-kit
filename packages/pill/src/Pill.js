import React from 'react'
import PropTypes from 'prop-types'
import cls from 'classnames'

import { prefix } from '@talixo/commons'

const name = prefix('pill')

/**
 * Component which represents Pill.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @returns {React.Element}
 */
function Pill (props) {
  const { children, className, color, style, variant, ...passedProps } = props

  return (
    <span
      className={cls(name, className, {
        [`${name}--${color}`]: color !== undefined,
        [`${name}--${variant}`]: variant !== undefined
      })}
      {...passedProps}
      style={style}
    >
      {children}
    </span>
  )
}

Pill.propTypes = {
  /** Additional class name */
  className: PropTypes.string,

  /** Children */
  children: PropTypes.node,

  /** Color */
  color: PropTypes.string,

  /** CSS Styles */
  style: PropTypes.object,

  /** Variants */
  variant: PropTypes.string
}

export default Pill
