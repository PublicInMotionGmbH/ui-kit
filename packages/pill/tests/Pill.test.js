import React from 'react'
import { shallow } from 'enzyme'

import Pill from '../src/Pill'

describe('<Pill />', () => {
  it('renders children correctly', () => {
    const wrapper = shallow(<Pill />)

    expect(wrapper).toMatchSnapshot()
  })

  it('handles `size` properly', () => {
    const wrapper = shallow(
      <Pill color='blue'>
        Label
      </Pill>
    )

    expect(wrapper.props().className).toMatch(/(^| )talixo-pill--blue( |$)/)
  })

  it('handles `className` properly', () => {
    const wrapper = shallow(
      <Pill className='anything'>
        Label
      </Pill>
    )

    expect(wrapper.props().className).toMatch(/(^| )anything( |$)/)
  })

  it('handles different `span` props properly', () => {
    const wrapper = shallow(
      <Pill style={{ padding: 20 }}>
        Label
      </Pill>
    )

    expect(wrapper.props().style).toEqual({ padding: 20 })
  })
})
