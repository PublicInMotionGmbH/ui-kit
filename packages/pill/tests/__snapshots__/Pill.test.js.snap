// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill /> renders children correctly 1`] = `
<span
  className="talixo-pill"
/>
`;
